### PR TITLE
fix incompatible AAD token

### LINKED ISSUE #86410: AAD token obtained by kubectl has spn prefix which is incompatible with AAD on-behalf-of flow
**What happened**:
In the [code path](https://github.com/kubernetes/kubernetes/search?q=NewOAuthConfig&unscoped_q=NewOAuthConfig) to acquire AAD token, because of the inclusion of the API version in the URLs for authorization, the returned token is not compatible with many AAD scenarios like OIDC and On-Behalf-Of flow.

There is a long list of issues documenting this problem:
https://github.com/Azure/go-autorest/issues/277
https://github.com/Azure/go-autorest/issues/321
https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/issues/128
https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/57
https://github.com/Azure/aks-engine/issues/376

The fix is pretty simple, use [this api](https://github.com/Azure/go-autorest/blob/2913f263500c4a5b23dada1b46ccd22ac972315f/autorest/adal/config.go#L56) with `apiVersion` set to `nil`.

This incompatible behavior impacts 
https://github.com/appscode/guard/issues/131
https://github.com/appscode/guard/issues/235

**What you expected to happen**:
Follow the [official documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-protocols-oauth-code#use-the-authorization-code-to-request-an-access-token) to acquire AAD token 

**How to reproduce it (as minimally and precisely as possible)**:
use kubectl to get AAD token, paste token into https://jwt.ms/ and verify the audience claim

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): all versions
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by weinong**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
It fixes issue #86410 where AAD token obtained by kubectl is incompatible with on-behalf-of flow and oidc. The `audience` claim before this fix has `spn:` prefix. After this fix, `spn:` prefix is omitted.

**Which issue(s) this PR fixes**:
Fixes #86410

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fixes issue where AAD token obtained by kubectl is incompatible with on-behalf-of flow and oidc.
The audience claim before this fix has "spn:" prefix. After this fix, "spn:" prefix is omitted.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
Looks good.

Could you sign CLA following https://github.com/kubernetes/kubernetes/pull/86412#issuecomment-567326168 and add a release note for this change? We need cherry-pick this to old branches.

**Comment by andyzhangx** [comment]
it said: "do-not-merge/invalid-commit-message", pls change your commit message. @weinong

**Comment by andyzhangx** [comment]
let's wait for the azure test result, will approve if it passes

**Comment by weinong** [comment]
@feiskyer thanks. signed cla and fixed release note

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
/assign @mikedanese

@mikedanese could you help to approve the changes? Thanks

**Comment by feiskyer** [comment]
e2e-gce tests are flacky: [Flaky Test] In-tree Volumes #86318

**Pull request merged**
